### PR TITLE
Add vit

### PR DESCRIPTION
**Pull request by kevinyamauchi**
add a pytorch lightning module for training a vision transformer autoencoder

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -38,6 +38,9 @@ install_requires =
     qtpy
     scikit-image
     scipy
+    tensorboard
+    torch
+    torchvision
```
Comment (leopold-franz) [inline setup.cfg:43]: monai is missing

File: src/morphospaces/networks/constants.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# dataset key for the ViT autoencoder augmentation views 1 and 2
+VIT_AC_AUG_VIEW_1_KEY = "image_1"
+VIT_AC_AUG_VIEW_2_KEY = "image_2"
+
+# dataset key for the Vit autoencoder ground truth image
+VIT_AC_GT_KEY = "gt_image"
```
Comment (leopold-franz) [inline src/morphospaces/networks/constants.py:6]: Why are you exporting these variables?

File: examples/pretrain_ssl.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+#
+# print("Total Number of Training Data Samples: {}".format(len(train_data)))
+# print(train_data)
+# print("#" * 10)
+# print("Total Number of Validation Data Samples: {}".format(len(val_data)))
+# print(val_data)
+# print("#" * 10)
+#
+# # Set Determinism
+# set_determinism(seed=123)
```
Comment (leopold-franz) [inline examples/pretrain_ssl.py:64]: Is the set_determinism not needed anymore? Where is this done now?

File: examples/pretrain_ssl.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    # make the checkpoint callback
+    best_checkpoint_callback = ModelCheckpoint(
+        save_top_k=1,
+        monitor="val_loss",
+        mode="min",
+        dirpath=logdir_path,
+        filename="vit-best",
+    )
+    last_checkpoint_callback = ModelCheckpoint(
+        save_last=True,
```
Comment (leopold-franz) [inline examples/pretrain_ssl.py:183]: From the documentation this seems to only save the last checkpoint and not the whole history of checkpoints. Therefore the code seems fine. I will double check this though once I run this on euler though :)

File: setup.cfg (hunk 1 above)
Comment (leopold-franz) [reply setup.cfg:43]: Monai will make torch and torchvision redundant i believe and you can even specify tensorboard as an additional dependency as monai[tensorboard]

**Review by leopold-franz — COMMENTED** [review]
I though I might as well leave my thoughts as comments. :) Thank you for porting this to pytorch lightning 💯

File: examples/pretrain_ssl.py (hunk 3 above)
Comment (kevinyamauchi) [reply examples/pretrain_ssl.py:64]: Good catch. We should use the pytorch lightning seed_everything function

https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.utilities.seed.html

File: examples/pretrain_ssl.py (hunk 4 above)
Comment (kevinyamauchi) [reply examples/pretrain_ssl.py:185]: it was a bit unclear to me from the docs. Thanks!

File: setup.cfg (hunk 1 above)
Comment (kevinyamauchi) [reply setup.cfg:43]: I will add monai. I prefer to keep torch and torchvision explicitly since we will directly import them.
